lib/pagination.js

```diff
@@ -1,41 +1,183 @@
+   *
+   * @param {number} page the page number to navigate to.
+   *
+   * @return {Promise<Object>} FHIR resources in a FHIR Bundle structure.
+   */
+  goToPage(page) {
+    const selectedPage = this.goToPageLink(page);
+
+    this.currentResults = selectedPage ? this.httpClient.get(selectedPage) : undefined;
+    return this.currentResults;
+  }
+
+  /**
+   * Return the current page of results.
+   *
    * @return {Promise<Object>} FHIR resources in a FHIR Bundle structure.
    */
-  nextPage(results) {
-    const nextLink = results.link.find(link => link.relation === 'next');
-    return nextLink ? this.httpClient.get(nextLink.url) : undefined;
+  currentPage() {
+    return this.getAndSetCurrent(this.selfLink());
+  }
+
+  /**
+   * Return the next page of results.
+   *
+   * @return {Promise<Object>} FHIR resources in a FHIR Bundle structure.
+   */
+  nextPage() {
+    return this.getAndSetCurrent(this.nextLink());
   }
 
   /**
    * Return the previous page of results.
    *
-   * @param {object} results a bundle result of a FHIR search
+   * @return {Promise<Object>} FHIR resources in a FHIR Bundle structure.
+   */
+  prevPage() {
+    return this.getAndSetCurrent(this.prevLink());
+  }
+
+  /**
+   * Return the link for the next page of results.
+   *
+   * @return {String} The link for the next page of results.
+   */
+  nextLink() {
+    return this.getLink(/next/);
+  }
+
+  /**
+   * Return the link for the previous page of results.
+   *
+   * @return {String} link for the previous page of results.
+   */
+  prevLink() {
+    return this.getLink(/^prev(ious)?$/);
+  }
+
+  /**
+   * Return the link for the current page of results.
+   *
+   * @return {String} link for the current page of results.
+   */
+  selfLink() {
+    return this.getLink(/self/);
+  }
+
+  /**
+   * Return a link for the specified page of results based on available search parameters.
+   *
+   * @param {number} page the page number to navigate to.
+   *
+   * @return {String} link for the specified page of results.
+   */
+  goToPageLink(page) {
+    const { countParam, offsetParam } = this.paramNames;
+
+    this[offsetParam] = (page * this[countParam]) - this[countParam];
+    const paramNames = Object.keys(this.paramNames);
+
+    const pageLinkUrl = new URL(`${this.baseUrl}`);
+
+    paramNames.forEach((paramName) => {
+      const param = this.paramNames[paramName];
+      pageLinkUrl.searchParams.append(param, this[param]);
+    });
+
+    return pageLinkUrl.href;
+  }
+
+  /**
+   * Return results and set them to the current results.
+   *
+   * @param {String} link a link referring to a page of search results.
    *
    * @return {Promise<Object>} FHIR resources in a FHIR Bundle structure.
    */
-  prevPage(results) {
-    const prevLink = results.link.find(link => link.relation.match(/^prev(ious)?$/));
-    return prevLink ? this.httpClient.get(prevLink.url) : undefined;
+  getAndSetCurrent(link) {
```

Comment:
I'd rename this `setCurrent`. I think it's pretty normal for a setter to return the value that was set, and the fact that it does so doesn't need to be called out in its name. Maybe we could look into using getters and setters or something and come up with a solution that doesn't need `set` in the name either.